language/move-lang/tests/move_check/expansion/assign_non_simple_name.move

```diff
@@ -0,0 +1,32 @@
+address 0x42 {
+module X {
+    struct S {}
+}
+module M {
+    use 0x42::X;
+    use 0x42::X::S as X;
+
+    struct R {}
+    struct S<T> { f: T }
+
+
+    fun t() {
+        X::S = ();
+        Self::S<u64> = ();
+        Self::R = ();
+
+        0x42::X::S = ();
+        0x42::M::S<u64> = ();
+        0x42::M::R = ();
+
+        x<u64> = ();
+        S<u64> = ();
+
+        X = ();
+        S = ();
+        R = ();
+
+        Y = 0;
```

Comment:
Just as a general comment, I keep finding Move tests (not necessarily for the compiler) where I can't tell what the test was intended to check. Especially with expected output files, it is easy for things to evolve over time so that a test is no longer doing what it was originally designed to do. This "Y = 0" stands out to me here as a particular case where it would be good to have a comment in the .move file to explain what it is testing, e.g.:
// Check that assignment to a ModuleAccess that references an unbound type is reported as an error.